utils/initial_data_preparation.py

```diff
@@ -160,71 +188,82 @@ def is_complete_url(url):
         # Check if the url is not a string
         if not isinstance(url, str):
             return False
-
         # Example url: https://github.com/owner/repo
-        parts = url.rstrip("/").split("/")
-        return len(parts) >= 5
+        parts = url.split("/")
+        return len(parts) >= EXPECTED_URL_PARTS
 
     # Identify rows with incomplete URLs using the helper function
-    incomplete_urls = df[~df["repourl"].apply(is_complete_url)]
+    df["incomplete_url_flag"] = ~df["repourl"].apply(is_complete_url)
+    incomplete_count = df["incomplete_url_flag"].sum()
+    logger.info(
+        f"Filtering Out Incomplete URLs: Found {incomplete_count} incomplete " f"URLs."
```

Comment:
Are we actually removing rows with incomplete URLs? I thought we're now keeping these rows in the overall dataframe, however the rows are marked as having an incomplete URL for their repo URL.

Reply:
No we do not remove any rows at any points. I can change the function name to `marking_incomplete_urls`. I will need to make small changes in the test file `test_github_repo_request_local.py`.

Reply:
- Function name has been changed across the codebase
- test cases for this function has been modified